### PR TITLE
 added the possibility to lock a record

### PR DESCRIPTION
**Pull request by gennai**
#### PR description:

I have added the possibility to lock a record

#### PR validation:

tested with unitTest

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@gennai since this PR is affecting Online DQM mainly, could you make a 11_3_X PR version to test it at P5 machines? Next MWGR is planning to use 11_3_X, unless you don't plan to use it by then

**Comment by gennai** [comment]
I am already working on it, I was waiting this PR to be scrutinized to apply the same comments to the back porting

**Comment by gennai** [comment]
Backporting is #33684

Best,
S.

**Comment by jfernan2** [comment]
Thank you! @gennai

**Comment by jfernan2** [comment]
+1
Backport tested succesfully in Online DQM at P5

**Review by silviodonato — COMMENTED** [review]
I would also suggest (not related to this PR) to use `clone` 
```
process.dqmBeamMonitor = process.dqmFakeBeamMonitor.clone(
   OnlineMode = True,
  useLockRecords =  useLockRecords,
  [...]
)
```
instead of 
```
process.dqmBeamMonitor = process.dqmFakeBeamMonitor.clone()
process.dqmBeamMonitor.useLockRecords = useLockRecords
process.dqmBeamMonitor.OnlineMode = True
[...]
```

in the first case the mis-spelled variable name would give error. See for instance https://github.com/cms-sw/cmssw/pull/33046/files
and 
`When one customizes an existing parameter in clone(), Modifier.toModify(), or in assignment, explicit types on the right hand side should be avoided. ` in https://cms-sw.github.io/cms_coding_rules.html

File: DQM/BeamMonitor/plugins/BeamMonitor.cc [hunk 1]
```diff
@@ -137,6 +137,7 @@ BeamMonitor::BeamMonitor(const ParameterSet& ps)
   minNrVertices_ = ps.getParameter<ParameterSet>("PVFitter").getUntrackedParameter<unsigned int>("minNrVerticesForFit");
   minVtxNdf_ = ps.getParameter<ParameterSet>("PVFitter").getUntrackedParameter<double>("minVertexNdf");
   minVtxWgt_ = ps.getParameter<ParameterSet>("PVFitter").getUntrackedParameter<double>("minVertexMeanWeight");
+  useLockRecords_ = ps.getUntrackedParameter<bool>("useLockRecords");
```
[the remark this answers is not in the document]
Comment (gennai) [reply DQM/BeamMonitor/plugins/BeamMonitor.cc:140]: Hi @silviodonato , thanks for the tips. Regarding the clone comment if you wish I can change it and commit. Let me know if I should do it or not.
For the untracked vs tracked parameters I have simply followed what was already there for the others, I have no strong feeling about it.

**Comment by silviodonato** [comment]
+1
I got confused with FakeBeamMonitor.cc that was made recently.
So I forward the comments to @cms-sw/dqm-l2. 
Ideally the untracked parameters should be used only for those parameters that does not change the physics result.

**Comment by jfernan2** [comment]
@silviodonato I can create a general PR modifying all the python config files in DQM packages if needed, including those Online. 

About tracked/untracked parameters, the Provenance is lost in DQM plain root files which are the final endproduct of DQM modules after harvesting, right? Though it might be tracked up in the EDM root files for intermediate MEs not produced in Harvesting step

**Comment by silviodonato** [comment]
> @silviodonato I can create a general PR modifying all the python config files in DQM packages if needed, including those at Online.

It is not urgent, but it would be a nice style improvement. Reco is making many PRs in that direction (eg. https://github.com/cms-sw/cmssw/pull/30947)

> About tracked/untracked parameters, the Provenance is lost in DQM plain root files which are the final endproduct of DQM modules after harvesting, right? Though it might be tracked up in the EDM root files for intermediate MEs not produced in Harvesting step

Ok, good point. I thought that it might be important to know if the module ran with useLockRecords True or False but, as you said, we cannot have this piece of information in plain DQM ROOT files.

**Comment by jfernan2** [comment]
OK, I will take care of this cloning restyling for master branch in the near future. Thanks!

**Comment by gennai** [comment]
About the tracked or untracked parameter, I do not think It would be interest anyway in keep track of it. It is something which does not affect the beamspot computation per se.

**Pull request merged**